src/core/Akka.Remote/Transport/AkkaPduCodec.cs

```diff
@@ -228,14 +228,19 @@ protected AkkaPduCodec(ActorSystem system)
         /// <returns>TBD</returns>
         public virtual ByteString EncodePdu(IAkkaPdu pdu)
         {
-            ByteString finalBytes = null;
-            pdu.Match()
-                .With<Associate>(a => finalBytes = ConstructAssociate(a.Info))
-                .With<Payload>(p => finalBytes = ConstructPayload(p.Bytes))
-                .With<Disassociate>(d => finalBytes = ConstructDisassociate(d.Reason))
-                .With<Heartbeat>(h => finalBytes = ConstructHeartbeat());
-
-            return finalBytes;
+            switch (pdu)
+            {
+                case Payload p:
```

Comment:
Change the order in which we filter for events in Akka.Remote serialization; `Payload` makes up the vast majority of network traffic, since that's all of the messages that aren't part of the Akka.Remote association protocol itself. `Payload` was previously handled second, so we always had one case we needed to fall through each time.

`Heartbeat` is the second most frequently used message type, since these are emitted once per second.